packages/snap-store-mobx/src/Search/Stores/FacetStore.test.ts

```diff
@@ -222,6 +224,349 @@ describe('Facet Store', () => {
+				expect(facets.length).toBe(2);
+			});
+
+			it('can be set to trim a specific facet', () => {
+				const settingsConfig = {
+					...searchConfig,
+					settings: {
+						facets: {
+							trim: false,
+							fields: {
+								color_family: {
+									trim: true,
+								},
+							},
+						},
+					},
+				};
+
+				const searchData = mockData.searchMeta('settings.trim');
+
+				const facets = new FacetStore(settingsConfig, services, storageStore, searchData.facets, searchData.pagination, searchData.meta);
+				expect(facets.length).toBe(3);
+			});
+		});
+
+		describe('Pin Filtered Facets', () => {
+			it('can be set to pin all selected facet values to the top of the values array', () => {
+				const settingsConfig = {
+					...searchConfig,
+					settings: {
+						facets: {
+							pinFiltered: true,
+						},
+					},
+				};
+
+				const searchData = mockData.searchMeta('settings.pinFiltered');
+
+				const facets = new FacetStore(settingsConfig, services, storageStore, searchData.facets, searchData.pagination, searchData.meta);
+				expect(facets.length).toBe(2);
+				facets.forEach((facet) => {
+					facet.values.forEach((value, index) => {
+						if (index == 0) {
+							expect(value).toHaveProperty('filtered', true);
+						}
+					});
+				});
+			});
+
+			it('can be set to NOT pin all selected facet values to the top of the values array', () => {
+				const settingsConfig = {
+					...searchConfig,
+					settings: {
+						facets: {
+							pinFiltered: false,
+						},
+					},
+				};
+
+				const searchData = mockData.searchMeta('settings.pinFiltered');
+
+				const facets = new FacetStore(settingsConfig, services, storageStore, searchData.facets, searchData.pagination, searchData.meta);
+				expect(facets.length).toBe(2);
+				facets.forEach((facet) => {
+					expect(facet.values[2]).toHaveProperty('filtered', true);
+				});
+			});
+
+			it('can be set to NOT pin all selected facet values to the top of the values array for a specific facet', () => {
+				const settingsConfig = {
+					...searchConfig,
+					settings: {
+						facets: {
+							pinFiltered: true,
+							fields: {
+								size: {
+									pinFiltered: false,
+								},
+							},
+						},
+					},
+				};
+
+				const searchData = mockData.searchMeta('settings.pinFiltered');
+
+				const facets = new FacetStore(settingsConfig, services, storageStore, searchData.facets, searchData.pagination, searchData.meta);
+				expect(facets.length).toBe(2);
+				facets.forEach((facet) => {
+					if (facet.field == 'size') {
+						expect(facet.values[2]).toHaveProperty('filtered', true);
+					} else {
+						expect(facet.values[0]).toHaveProperty('filtered', true);
+					}
+				});
+			});
+
+			it('can be set to pin all selected facet values to the top of the values array for a specific facet', () => {
+				const settingsConfig = {
+					...searchConfig,
+					settings: {
+						facets: {
+							pinFiltered: false,
+							fields: {
+								size: {
```

Comment:
Actually, I don't think this is an issue in this case. The tests reveal this works when enabled/disabled for this facet.

In the mock data I am only ever selecting one value (the 3rd and last one) from each facet. `pinFiltered` moves the `filtered` values to the top whereas the normal behavior without it would to retain the order of the facet values.